ml4gw/distributions.py

```diff
@@ -108,3 +108,15 @@ def __call__(self, N: int) -> torch.Tensor:
         samples = self.x_min ** (-self.alpha + 1) - u
         samples = torch.pow(samples, -1.0 / (self.alpha - 1))
         return samples
+
+
+class ParameterSampler(torch.nn.Module):
```

Comment:
Not sure this makes sense to implement here, since it's a higher-level API leveraging these more fundamental `Callable`s. Think it would make more sense to implement in the waveform submodule itself, or if we think this is something we'll want to use more generally creating a `sampling` submodule or something and having this and `distributions.py` live there. My inclination is to go with the former until we have a more general use case, but up for debate